jmdaemon/jmdaemon/irc.py

```diff
@@ -139,16 +138,14 @@ def build_irc(self):
             #str() casts needed else unicode error
             torEndpoint = TCP4ClientEndpoint(reactor, str(self.socks5_host),
                                              self.socks5_port)
-            ircEndpoint = SOCKS5ClientEndpoint(str(self.serverport[0]),
-                                               self.serverport[1], torEndpoint)
             if self.usessl.lower() == 'true':
-                ctx = ClientContextFactory()
-                tlsEndpoint = TLSWrapClientEndpoint(ctx, ircEndpoint)
-                myRS = ClientService(tlsEndpoint, factory)
-                myRS.startService()
+                use_tls = True
             else:
-                myRS = ClientService(ircEndpoint, factory)
-                myRS.startService()
+                use_tls = False
+            ircEndpoint = TorSocksEndpoint(torEndpoint, self.serverport[0],
+                                           self.serverport[1], tls=use_tls)
```

Comment:
Will that actually make the underlying connection be TLS-wrapped or only the connection to the tor socks server? The documentation on this method is unfortunately very poor.

Reply:
I think so, maybe try connecting to a TLS only IRC server port and see what happens.